packages/core/src/RenderingEngine/vtkClasses/vtkStreamingOpenGLTexture.js

```diff
@@ -239,6 +176,78 @@ function vtkStreamingOpenGLTexture(publicAPI, model) {
   publicAPI.setUpdatedFrame = (frameIndex) => {
     model.updatedFrames[frameIndex] = true;
   };
+
+  const superModified = publicAPI.modified;
+  publicAPI.modified = () => {
+    // Really important piece here that wasted a lot of time from me
+    // always make sure to call the super method first
+    superModified();
+    const volume = cache.getVolume(model.volumeId);
+    const imageIds = volume.imageIds;
+
+    for (let i = 0; i < imageIds.length; i++) {
+      model.updatedFrames[i] = true;
+    }
+  };
+
+  publicAPI.hasUpdatedFrames = () => {
+    return model.updatedFrames.some((frame) => frame);
+  };
+
+  function updateTextureImagesUsingVoxelManager(updatedFrames) {
+    const volume = cache.getVolume(model.volumeId);
+    const imageIds = volume.imageIds;
+    for (let i = 0; i < updatedFrames.length; i++) {
+      if (updatedFrames[i]) {
+        // find the updated frames
+        const image = cache.getImage(imageIds[i]);
+
+        if (!image) {
+          continue;
```

Comment:
Might drop a warn level message here, as this might indicate something got evicted before it could be rendered.